gcloud/storage/acl.py

```diff
@@ -345,78 +346,84 @@ def get_entities(self):
         self._ensure_loaded()
         return list(self.entities.values())
 
-    def reload(self):
-        """Reload the ACL data from Cloud Storage.
+    @property
+    def connection(self):
+        """Compute the connection for API requests for this ACL.
 
         This is a virtual method, expected to be implemented by subclasses.
 
-        :raises: :class:`NotImplementedError`
+        :raises: :class:`NotImplementedError` if ``_connection`` attribute
+                 is not set on the instance.
+
         """
+        # Allow override for testing
+        connection = getattr(self, '_connection', None)
```

Comment:
Because this PR wasn't aimed at removing `self.connection` usage?  Anyway, having the base class define a property which raises unconditionally makes it impossible to test the "hoisted" methods (which were entirely redundant, except for paths).